r2-core/src/main/java/com/linkedin/r2/message/QueryTunnelUtil.java

```diff
@@ -408,6 +421,12 @@ else if(existingQuery.isEmpty())
     }
     requestBuilder.setEntity(entity);
     requestBuilder.setHeaders(h);
+
+    String overrideHeader = request.getHeader(HEADER_METHOD_OVERRIDE);
```

Comment:
This header is first set on line 181; shall we move this check there?  This is to make sure bad method string is not saved into the headers at all.

Reply:
I don't want to only validate in encode since an alternate client can still send invalid content. I've refactored the validation into its own method so we can validate in both locations.